src/parser.rs

```diff
@@ -140,15 +137,11 @@ fn search_for_attr(
         } else {
             match cur_node_value.kind() {
                 NODE_ATTR_SET => {
-                    if is_match {
-                        result.push((cur_node_value, path, real_depth));
-                    } else {
-                        let cur_node_casted = AttrSet::cast(cur_node_value).unwrap();
-                        cur_node_casted.entries().for_each(|entry| {
-                            stack.insert(0, (entry, path.clone()));
-                            remaining_items_cur += 1;
-                        });
-                    }
+                    let cur_node_casted = AttrSet::cast(cur_node_value).unwrap();
+                    cur_node_casted.entries().for_each(|entry| {
+                        stack.push((entry, path.clone(), real_depth));
+                        remaining_items_cur += 1;
+                    });
```

Comment:
```suggestion
                    let orig_stack_size = stack.len();
                    stack.extend(cur_node_casted.entries().map(|entry| (entry, path.clone(), real_depth)));
                    remaining_items_cur += stack.len() - orig_stack_size;
```

Reply:
additionally, this fixes #10.

Reply:
Changed! Also deleted remaining_items_* since they're not needed anymore. Depth only makes sense in the context of BFS.